pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go

```diff
@@ -61,6 +61,15 @@ func (l *Lifecycle) deploy(obj *v3.PipelineExecution) error {
 		return errors.Wrapf(err, "Error create secret")
 	}
 
+	apikey, err := l.systemAccountManager.GetOrCreateProjectSystemToken(projectID)
+	if err != nil {
+		return err
+	}
+	secret = getAPIKeySecret(nsName, apikey)
+	if _, err := l.secrets.Create(secret); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
Do we create secret for each deploy? Should we recycle them?

Reply:
Its lifecyle is the same as pipeline components.

Reply:
I think you should use lister to get secret from cache first, if it doesn't exist, then create it.

Reply:
Probably better if we find the dedicated namespace exists at the begining, we skip the whole deploy stuff. Any missing part would be addressed in upgrade logic.

Reply:
Updated